mlir/lib/Conversion/GPUCommon/GPUOpsLowering.cpp

```diff
@@ -498,7 +498,7 @@ LogicalResult GPUPrintfOpToVPrintfLowering::matchAndRewrite(
   for (auto [index, arg] : llvm::enumerate(args)) {
     Value ptr = rewriter.create<LLVM::GEPOp>(
         loc, LLVM::LLVMPointerType::get(arg.getType()), tempAlloc,
-        ArrayRef<LLVM::GEPArg>{0, index});
+        ArrayRef<LLVM::GEPArg>{0, static_cast<int32_t>(index)});
```

Comment:
Similarly - if the warning isn't catching real bugs, maybe it's not worth keeping enabled?

Reply:
will revert(or close PR)